### ISSUE TITLE
gluestick app won't start when run with Node v7.0.0

### ISSUE THREAD

**Issue by chefnobody** [TrueCar/gluestick#322]
Using:

```
Aarons-MacBook-Pro:gluestick aconnolly$ node -v
v7.0.0
Aarons-MacBook-Pro:gluestick aconnolly$ npm -v
3.10.8
```

And running `npm install` from a freshly forked `gluestick` repo results in the following errors:

```
Aarons-MacBook-Pro:gluestick aconnolly$ npm install

> v8-debug@0.7.7 preinstall /Users/aconnolly/workspace/gluestick/node_modules/.staging/v8-debug-c1f18b53
> node -e 'process.exit(0)'


> v8-debug@0.7.7 install /Users/aconnolly/workspace/gluestick/node_modules/v8-debug
> node-pre-gyp install --fallback-to-build

node-pre-gyp ERR! Tried to download(404): https://node-inspector.s3.amazonaws.com/debug/v0.7.7/node-v51-darwin-x64.tar.gz 
node-pre-gyp ERR! Pre-built binaries not found for v8-debug@0.7.7 and node@7.0.0 (node-v51 ABI) (falling back to source compile with node-gyp) 
  CXX(target) Release/obj.target/debug/src/debug.o
../src/debug.cc:38:27: error: too few arguments to function call, expected at least 2, have 1
        v8::Debug::Call(fn);
        ~~~~~~~~~~~~~~~   ^
/Users/aconnolly/.node-gyp/7.0.0/include/node/v8-debug.h:198:3: note: 'Call' declared here
  static MaybeLocal<Value> Call(Local<Context> context,
  ^
../src/debug.cc:46:61: error: too few arguments to function call, single argument 'isolate' was not specified
        Isolate* debug_isolate = v8::Debug::GetDebugContext()->GetIsolate();
                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
/Users/aconnolly/.node-gyp/7.0.0/include/node/v8-debug.h:248:3: note: 'GetDebugContext' declared here
  static Local<Context> GetDebugContext(Isolate* isolate);
  ^
../src/debug.cc:61:67: error: too few arguments to function call, single argument 'isolate' was not specified
        Local<Context> debug_context = v8::Debug::GetDebugContext();
                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
/Users/aconnolly/.node-gyp/7.0.0/include/node/v8-debug.h:248:3: note: 'GetDebugContext' declared here
  static Local<Context> GetDebugContext(Isolate* isolate);
  ^
../src/debug.cc:66:54: error: too few arguments to function call, single argument 'isolate' was not specified
          debug_context = v8::Debug::GetDebugContext();
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
/Users/aconnolly/.node-gyp/7.0.0/include/node/v8-debug.h:248:3: note: 'GetDebugContext' declared here
  static Local<Context> GetDebugContext(Isolate* isolate);
  ^
4 errors generated.
make: *** [Release/obj.target/debug/src/debug.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/aconnolly/workspace/gluestick/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Darwin 15.6.0
gyp ERR! command "/usr/local/bin/node" "/Users/aconnolly/workspace/gluestick/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/Users/aconnolly/workspace/gluestick/node_modules/v8-debug/build/debug/v0.7.7/node-v51-darwin-x64/debug.node" "--module_name=debug" "--module_path=/Users/aconnolly/workspace/gluestick/node_modules/v8-debug/build/debug/v0.7.7/node-v51-darwin-x64"
gyp ERR! cwd /Users/aconnolly/workspace/gluestick/node_modules/v8-debug
gyp ERR! node -v v7.0.0
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
node-pre-gyp ERR! build error 
```

It appears there are compatibility problems with node v7.0.0 and `gluestick`.

**Comment by ottoinfo** [comment]
@chefnobody optional workaround [NVM Installation](https://gist.github.com/ottoinfo/adbee2352a48290de8234904f9b1e7c8)

**Comment by zamotany** [comment]
From what I've investigated it looks like V8 started deprecation lifecycle (some time ago) on some functions and then removed them, but node-inspector have not incorporated this changes.  The reason why it's failing is that node-inspector is calling functions that doesn't exist both in newer version of node and V8.

If we can force usage of at least version 6.3.0 of node, we can use built in inspector.

**Comment by toddw** [comment]
Thanks @zamotany!

> If we can force usage of at least version 6.3.0 of node, we can use built in inspector.

I think it is reasonable to require _at least_ version 6.3.0 for debugging using the inspector. `node-inspector` never really worked that great even on the best days. The built in debugger works great!

Maybe that will be done as part of #406

**Comment by zamotany** [comment]
@toddw you mean node debugger or `node --inspect` that allows to debug in Chrome Dev Tools??
I need to clarify some things: you want to be able to use node debugger and connect to it via VS Code/WebStrorm/Atom or to use built in node inspector that translates **Node Debug Protocol** to **Chrome DevTools Protocol** allowing to debug in Chrome Dev Tools?

**Comment by toddw** [comment]
I was talking about 'node -- inspect'

**Comment by toddw** [comment]
To be more clear: there is a project called node-inspector which never really worked well for us but the built in version of 'node -- inspect' works great

**Comment by zamotany** [comment]
Ok, so `node --inspect` work great when debugging server, but as I wrote here:
https://github.com/TrueCar/gluestick/issues/406
I could not debug test, mainly because `debugger` statement's did not trigger break.
Did you successfully debug test with `node --inspect`??

**Comment by toddw** [comment]
I believe that is broken in generated apps but it works when debugging tests in the GlueStick repo. I'll do some research on this one and get back to you

**Comment by zamotany** [comment]
Debugging GlueStick test with `node --inspect` works using mocha in node or in browser with karma?

**Comment by toddw** [comment]
I'm pretty sure it works with mocha in node. I believe you can pass it as an option to _mocha_ directly.

**Comment by zamotany** [comment]
OK, so I quickly tested if `debugger` statement work with `node --inspector` and they work but not in with Jest. That maybe connected to the fact that Jest test with `debugger` statement are required dynamically.

**Comment by toddw** [comment]
This is what I use to run tests in the GlueStick repo: `npm test --- --inspect --debug-brk`. Then I'll usually just type `debugger` on the line I want to break

**Comment by toddw** [comment]
This is likely resolved with #445
